### PR TITLE
Fixed/implemented metanorma examples

### PR DESCRIPTION
**Pull request by suleman-uzair**
Extracted examples from metanorma repos(metioned below) and added/updated Parslet rules for failing scenarios.

1. mn-samples-iec
2. mn-samples-itu
3. mn-samples-bipm
4. mn-samples-ieee-private
5. mn-samples-iho
6. mn-samples-iso
7. mn-samples-ogc
8. mn-samples-jcgm
9. mn-samples-nist

### CODE REVIEW

File: lib/plurimath/math/function/mod.rb [hunk 1]
```diff
@@ -7,8 +7,8 @@ module Math
     module Function
       class Mod < BinaryFunction
         def to_asciimath
-          first_value = parameter_one.to_asciimath if parameter_one
-          second_value = parameter_two.to_asciimath if parameter_two
+          first_value = wrapped(parameter_one) if parameter_one
+          second_value = wrapped(parameter_two) if parameter_two
```
Comment (HassanAkbar) [inline lib/plurimath/math/function/mod.rb:11]: The first line in the `wrapped` function is `return "" unless field`. Do we need an `if` condition here?

File: lib/plurimath/math/function/unary_function.rb [hunk 2]
```diff
@@ -16,30 +16,57 @@ def ==(object)
...
         def to_latex
-          first_value = "{#{parameter_one.to_latex}}" if parameter_one
+          first_value = "{#{latex_value}}" if parameter_one
           "\\#{class_name}#{first_value}"
         end
 
+        def latex_value
+          case parameter_one
+          when Array
+            parameter_one&.map { |value| value&.to_latex }.join
```
Comment (HassanAkbar) [inline lib/plurimath/math/function/unary_function.rb:58]:
```suggestion
            parameter_one&.compact&.map(&:to_latex)&.join
```

File: lib/plurimath/utility.rb [hunk 3]
```diff
@@ -110,7 +108,7 @@ def rpr_element(wi_tag: false)
       end
 
       def update_nodes(element, nodes)
-        nodes.each { |node| element << node unless node.nil? }
+        nodes&.each { |node| element << node unless node.nil? }
```
Comment (HassanAkbar) [inline lib/plurimath/utility.rb:111]:
```suggestion
        nodes&.compact&.each { |node| element << node }
```

**Review by HassanAkbar — CHANGES_REQUESTED** [review]

File: lib/plurimath/utility.rb (hunk 3 above)
Comment (suleman-uzair) [reply lib/plurimath/utility.rb:111]: *`nodes`* variable contains **Ox::Element** object which supports *`each`* but doesn't support *`compact`*.

File: spec/plurimath/latex/metanorma_exampes_spec.rb [hunk 4]
```diff
@@ -0,0 +1,199 @@
+require_relative "../../../lib/plurimath/math"
+
```
Comment (ronaldtse) [inline spec/plurimath/latex/metanorma_exampes_spec.rb:2]: The filename of this file contains a typo in `exampes`

File: spec/plurimath/latex/metanorma_exampes_spec.rb [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+RSpec.describe Plurimath::Latex::Parser do
```
Comment (ronaldtse) [inline spec/plurimath/latex/metanorma_exampes_spec.rb:3]: @suleman-uzair for all imported specs/tests, can you label the source of the formula? e.g. `metanorma/mn-samples-iso:{path name}`?

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: spec/plurimath/latex/metanorma_exampes_spec.rb (hunk 5 above)
Comment (ronaldtse) [reply spec/plurimath/latex/metanorma_exampes_spec.rb]: Thanks, I've also updated it.

**Comment by ronaldtse** [comment]
Thank you @suleman-uzair ! Could you help resolve the remaining comments from @HassanAkbar and then merge this?

**Review by ronaldtse — APPROVED** [review]

**Pull request merged**